### PR TITLE
Added type3 iface

### PR DESCRIPTION
**Pull request by dlaube**

### CODE REVIEW

**Comment by marineam** [comment]
LGTM, @crawford?

File: network/packet.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func parseNetwork(netdata packet.NetworkData, nameservers []net.IP) ([]Interface
 	for _, iface := range netdata.Interfaces {
 		if iface.Name != "chassis0" && iface.Name != "ipmi0" {
 			bond.slaves = append(bond.slaves, iface.Name)
-			if iface.Name == "enp1s0f0" {
+			if iface.Name == "enp1s0f0" || iface.Name == "ens20" {
```
Comment (crawford) [inline network/packet.go]: Do the machines only ever have one of these two interfaces?

**Comment by stresler** [comment]
Working on testing this internally, will comment when we're ready on our side.

File: network/packet.go (hunk 1 above)
Comment (stresler) [inline network/packet.go]: We're going to run through quite a few of the systemd-udev naming conventions, but they all appear to order alphabetically, so I think we can remove this check and go alphabetical.

**Comment by stresler** [comment]
I'm not sure why this is failing. It's passing locally and Travis isn't saying why it's exiting 1.

**Comment by crawford** [comment]
@stresler the error output isn't fantastic, but it's not happy with the formatting of network/packet.go.

File: network/packet.go [hunk 2]
```diff
@@ -98,33 +98,23 @@ func parseNetwork(netdata packet.NetworkData, nameservers []net.IP) ([]Interface
...
+				name: iface.Name,
+				config: configMethodStatic{
+					nameservers: nameservers,
+				},
+				children: []networkInterface{&bond},
+				configDepth: index,
+			},
 		}
+
+		interfaces = append(interfaces, &p)
```
Comment (crawford) [inline network/packet.go]: `p`'s declaration can be flattened into this line.

Comment (stresler) [inline network/packet.go]: I'm sorry, I'm not really sure what you mean by this. Do you want me to declare p in the append function?  Doesn't it need to be a & type pointer?

Comment (crawford) [inline network/packet.go]: In go, it is valid to write:

```go
interfaces = append(interfaces, &physicalInterface{
    logicalInterface: logicalInterface{
        name: iface.Name,
        config: configMethodStatic{
            nameservers: nameservers,
        },
        children: []networkInterface{&bond},
        configDepth: index,
    },
})
```

Comment (stresler) [inline network/packet.go]: mind...blown. Ok - will have that in shortly. 

Is that defining the struct in memory and making a pointer at the same time? That's handy.

Comment (crawford) [inline network/packet.go]: Yep. Go's garbage collection makes this possible (though, this functionality doesn't strictly require garbage collection; a sufficiently smart compiler could also allow this).

**Comment by crawford** [comment]
LGTM. Please squash and prefix the commit with "network: ".

**Comment by stresler** [comment]
Let us run one more round of tests on our hardware, and then will do.

**Comment by crawford** [comment]
@stresler bump.

**Pull request merged**